amr-wind/incflo_advance.cpp

```diff
@@ -52,27 +36,37 @@ void incflo::advance()
         field.advance_states();
         field.state(amr_wind::FieldState::Old).fillpatch(m_time.current_time());
     }
+}
 
+void incflo::pre_advance_stage2()
+{
+    BL_PROFILE("amr-wind::incflo::pre_advance_stage2");
     for (auto& pp: m_sim.physics())
         pp->pre_advance_work();
+}
 
+/** Advance simulation state by one timestep
+ *
+ *  Performs the following actions at a given timestep
+ *  - Compute \f$\Delta t\f$
+ *  - Advance all computational fields to new timestate in preparation for time integration
+ *  - Call pre-advance work for all registered physics modules
+ *  - For Godunov scheme, advance to new time state
+ *  - For MOL scheme, call predictor corrector steps
+ *  - Perform any post-advance work
+ *
+ *  Much of the heavy-lifting is done by incflo::ApplyPredictor and
+ *  incflo::ApplyCorrector. Please refer to the documentation of those methods
+ *  for detailed information on the various equations being solved.
+ *
+ * \callgraph
+ */
+void incflo::advance()
+{
+    BL_PROFILE("amr-wind::incflo::Advance");
     ApplyPredictor();
 
-    if (!m_use_godunov) {
-        vel.state(amr_wind::FieldState::New).fillpatch(m_time.current_time());
-        for (auto& eqn: scalar_eqns()) {
```

Comment:
should we keep this around?

Reply:
regression tests pass so maybe this was not necessary?

Reply:
Yeah, the logic is now handled in post solve ops.